### PR TITLE
Chore: Ensuring error messages appear before successes

### PR DESCRIPTION
**Pull request by MikaelMayer**
In the IDE, when there are many assertions on the same range, hovering over them reveals information about them in the order they were verified. However, the information most relevant to users is the one about verifications errors. This PR moves them first.

I will add tests shortly in case no test fails.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs [hunk 1]
```diff
@@ -150,18 +154,29 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
...
             }
 
             assertionIndex++;
           }
         }
 
+        var biggerResourceCountFirst =
+          Comparer<(string content, long resources)>.Create(
+          (left, right) =>
+            (int)(right.resources / 1000000) - (int)(left.resources / 1000000)
```
Comment (atomb) [inline Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs:173]: Is there any need to do this division just to compare resource counts?

**Review by atomb — CHANGES_REQUESTED** [review]
This seems great to me, though there's one place where I think it could be simplified.

File: Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs (hunk 1 above)
Comment (MikaelMayer) [reply Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs]: Good question, the comparison metric requires an "int", when the resource count is a long. I have seen resource counts exceeding the range of an int in practice and display a negative result, but they should be displayed first.
I divided by one million to ensure we were practically within the int range, but I decided to do better. In the newest version, I'm mapping algorithmically long to int and added a comment for that.

Comment (atomb) [reply Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs]: I think the only significance of the `int` result is whether it's less than, equal to, or greater than zero. So I think `right.resources.CompareTo(left.resources)` would work.

Comment (MikaelMayer) [reply Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs]: Oh yeah did not know this method existed, and I could have implemented it myself. Thanks for the simplification.

**Review by atomb — APPROVED** [review]
This will be a nice help!

**Pull request merged**